packages/blade/src/components/Popover/__tests__/Popover.web.test.tsx

```diff
@@ -0,0 +1,281 @@
+        <Button>{buttonText}</Button>
+      </Popover>,
+    );
+    await waitForPosition();
+
+    // should not be opened by default
+    expect(queryByRole('dialog')).toBeInTheDocument();
+    expect(onChange).toHaveBeenCalledTimes(0);
+
+    // close after 300ms
+    fireEvent.click(getByRole('button', { hidden: true, name: 'Close' }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(onChange).toHaveBeenCalledTimes(1);
+    expect(onChange).toHaveBeenCalledWith(false);
+
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+
+    // open
+    fireEvent.click(getByRole('button', { name: buttonText }));
+    expect(queryByRole('dialog')).toBeInTheDocument();
+    expect(onChange).toHaveBeenCalledTimes(2);
+    expect(onChange).toHaveBeenCalledWith(true);
+  });
+
+  it('should work with controlled isOpen', async () => {
+    const buttonText = 'Click me';
+    const toggleButtonText = 'Toggle';
+
+    const ControlledExample = () => {
+      const [isOpen, setIsOpen] = React.useState(true);
+
+      return (
+        <>
+          <Popover
+            content="Hello world"
+            isOpen={isOpen}
+            onOpenChange={({ isOpen }) => setIsOpen(isOpen)}
+          >
+            <Button onClick={() => setIsOpen(true)}>{buttonText}</Button>
+          </Popover>
+          <Button onClick={() => setIsOpen((prev) => !prev)}>{toggleButtonText}</Button>
+        </>
+      );
+    };
+    const { getByRole, queryByRole } = renderWithTheme(<ControlledExample />);
+    await waitForPosition();
+
+    // should be opened by default
+    expect(queryByRole('dialog')).toBeInTheDocument();
+
+    // should close after 300ms
+    fireEvent.click(getByRole('button', { hidden: true, name: toggleButtonText }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+
+    // open again
+    fireEvent.click(getByRole('button', { name: buttonText }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299);
+    });
+
+    expect(queryByRole('dialog')).toBeInTheDocument();
+
+    // close again with close icon
+    fireEvent.click(getByRole('button', { hidden: true, name: 'Close' }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should have work with PopoverInteractiveWrapper', async () => {
+    jest.useFakeTimers();
+    const { getByTestId, queryByRole } = renderWithTheme(
+      <Popover content="Hello world">
+        <PopoverInteractiveWrapper>
+          <InfoIcon color="surface.action.icon.default.highContrast" size="medium" />
+        </PopoverInteractiveWrapper>
+      </Popover>,
+    );
+    const wrapper = getByTestId('popover-interactive-wrapper');
```

Comment:
Is this a fixed value for testID? We should not have a default value for the `testId` prop, it must be passed manually. Otherwise, It can cause unintentional behavior in case we have multiple usage of `PopoverInteractiveWrapper`.

Reply:
It has a default but people can override the testID prop if they want, sure removed.